### DISCUSSION TITLE
Nested color modes

### DISCUSSION THREAD

**Discussion by herrethan** [system-ui/theme-ui#2403] [Q&A]
Say you want to show a small portion of your site in dark mode while in light mode. Because nested `ThemeProvider`s will always inherit the root colorMode, doing this becomes tricky. Some discussion on that [here](https://github.com/system-ui/theme-ui/issues/2230).

However, digging around in the source code I found that NestedColorModeProvider creates a helper class that allows this. 

```js
// ... light themed stuff
<div className="theme-ui-dark">
  <ThemeProvider theme={{}}>
    {/* dark themed stuff */}
  </ThemeProvider>
</div>
```

My question is, is this safe to use? Since it's not documented the concern is how long this might be supported.

**Comment by pietrofxq** [comment]
I have the same question.. I think a workaround would be to create a wrapper for the provider, and pass the same theme object with a modified `initialColorModeName`. I am not sure how this will behave when actually changing the color mode though

**Reply by pietrofxq** [reply]
something like this seem to work:
```
export const ThemeProvider = ({
  theme = 'light',
  children,
}: ThemeProviderProps) => (
  <ThemeUIProvider
    theme={
      theme === 'light'
        ? defaultTheme
        : {
            ...defaultTheme,
            config: {
              initialColorModeName: 'dark',
            },
          }
    }
  >
    {children}
  </ThemeUIProvider>
)
```
